woocommerce/woocommerce-gutenberg-products-block
# Remove hydration hocs in favour of apiFetch Middlewares

Gutenberg provides an apiFetch middleware called `createPreloadingMiddleware` which accepts REST API data. This is very very similar to the hydration code we implemented, but more straightforward to use. 

This PR passes endpoint data to `createPreloadingMiddleware` and removes our custom HOC implementation. This also removes the need to handle stale cart data on page load (implemented in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3960) because `createPreloadingMiddleware` does not exhibit the same problem.

**Testing**

1. Go to the cart block page. Check there were no extra `fetch` requests to `wc/store/cart` in the network inspector.
2. Go to the checkout block page. Ensure there is no error and checkout works as normal.

Testing stale data:

- Go to cart block
- Change cart item quantities. Take note of the amounts.
- Browse to a different page
- Use the browser back button.
- Confirm amounts are still correct and not stale.

@opr confirmed this removes the need for the fix in #3960 
@Aljullu could also use your eyes since it touched a few parts of the mini-cart block
